src/overlays/actors/ovl_En_Wf/z_en_wf.c

```diff
@@ -137,81 +144,1338 @@ static ColliderCylinderInit D_80B37A5C = {
+        if (this->actor.speedXZ < 0.0f) {
+            this->actor.speedXZ += 0.05f;
+        }
+
+        this->isInvisible = false;
+    }
+
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 1, 4500, 0);
+
+    if (!EnWf_ChangeAction(globalCtx, this, 0) && SkelAnime_Update(&this->skelAnime)) {
+        if (this->actor.bgCheckFlags & 1) {
+            angleToWall = this->actor.wallYaw - this->actor.shape.rot.y;
+            angleToWall = ABS(angleToWall);
+
+            if ((this->actor.bgCheckFlags & 8) && (ABS(angleToWall) < 12000) && (this->actor.xzDistToPlayer < 120.0f)) {
+                EnWf_SetupTurnTowardsPlayer(this);
+            } else if (!EnWf_DodgeRanged(globalCtx, this)) {
+                if ((this->actor.xzDistToPlayer <= 80.0f) && !Actor_OtherIsTargeted(globalCtx, &this->actor) &&
+                    ((globalCtx->gameplayFrames % 8) != 0)) {
+                    EnWf_SetupFirstSlash(this);
+                } else if (Rand_ZeroOne() > 0.5f) {
+                    EnWf_SetupWait(this);
+                    this->actionTimer = (Rand_ZeroOne() * 5.0f) + 5.0f;
+                    this->headTilt = 30;
+                } else {
+                    EnWf_SetupBackflip(this);
+                }
+            }
+        }
+    }
+}
+
+void EnWf_SetupTurnTowardsPlayer(EnWf* this) {
+    f32 lastFrame = Animation_GetLastFrame(&gWolfosBackflippingAnim);
+
+    Animation_Change(&this->skelAnime, &gWolfosBackflippingAnim, -1.0f, lastFrame, 0.0f, ANIMMODE_ONCE, -3.0f);
+    this->actionTimer = 0;
+    this->isInvisible = false;
+    this->action = WOLFOS_ACTION_TURN_TOWARDS_PLAYER;
+    this->actor.speedXZ = 6.5f;
+    this->actor.velocity.y = 15.0f;
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_STAL_JUMP);
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    EnWf_SetupAction(this, EnWf_TurnTowardsPlayer);
+}
+
+void EnWf_TurnTowardsPlayer(EnWf* this, GlobalContext* globalCtx) {
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 1, 4000, 1);
+
+    if (this->actor.velocity.y >= 5.0f) {
+        //! @bug unk_4C8 and unk_4BC are used but not set (presumably intended to be feet positions like other actors)
+        func_800355B8(globalCtx, &this->unk_4C8);
+        func_800355B8(globalCtx, &this->unk_4BC);
+    }
+
+    if (SkelAnime_Update(&this->skelAnime) && (this->actor.bgCheckFlags & (1 | 2))) {
+        this->actor.world.rot.y = this->actor.shape.rot.y = this->actor.yawTowardsPlayer;
+        this->actor.shape.rot.x = 0;
+        this->actor.speedXZ = this->actor.velocity.y = 0.0f;
+        this->actor.world.pos.y = this->actor.floorHeight;
+
+        if (!Actor_OtherIsTargeted(globalCtx, &this->actor)) {
+            EnWf_SetupFirstSlash(this);
+        } else {
+            EnWf_SetupWait(this);
+        }
+    }
+}
+
+void EnWf_SetupReactToPlayer(EnWf* this) {
+    f32 lastFrame = Animation_GetLastFrame(&gWolfosBlockingAnim);
```

Comment:
```suggestion
void EnWf_SetupBlocking(EnWf* this) {
    f32 lastFrame = Animation_GetLastFrame(&gWolfosBlockingAnim);
```

I mean, every action "reacts to the player actions" in some way, and this is invoked when the player is near or attacking (as well as the action using the blocking animation)